ensime_shared/protocol.py

```diff
@@ -232,10 +232,10 @@ def handle_doc_uri(self, call_id, payload):
 
     def handle_completion_info_list(self, call_id, payload):
         """Handler for a completion response."""
+        self.log.debug('handle_completion_info_list: in')
         completions = payload["completions"]
-        self.log("handle_completion_info_list: in")
         self.suggestions = [completion_to_suggest(c) for c in completions]
-        self.log("handle_completion_info_list: {}".format(self.suggestions))
+        self.log.debug('handle_completion_info_list: %s', self.suggestions)
```

Comment:
Indeed that could cut down on the conditional statements. Whether creating and GCing those wrapper objects is significantly more expensive than (possibly cached) calls to `log.isEnabledFor` and worth the tradeoff for readability is up for debate, but we're really in the weeds of micro-optimization at that point 😁   I don't feel strongly either way.